src/Analytics.js

```diff
@@ -0,0 +1,202 @@
+				});
+			}
+			this.props.onPlay && this.props.onPlay(event);
+		}
+
+		/**
+		 * Event triggered everytime the user clicks pause
+		 * @param {NativeEvent} event
+		 */
+		onPause(event) {
+			this.onEvent({'type': PlayerEventTypes.ON_PAUSE});
+			this.props.onPause && this.props.onPause(event);
+		}
+
+		/**
+		 * Event triggered when the video ends
+		 * @param {NativeEvent} event
+		 */
+		onEnd(event) {
+			this.onEvent({'type': PlayerEventTypes.ON_END});
+			this.props.onEnd && this.props.onEnd(event);
+		}
+
+
+		/**
+		 * Event triggered as the stream progress.
+		 * @param {NativeEvent} event
+		 * @param {number} event.currentTime - The current time of the video
+		 * @param {number} event.duration - The total duration of the video
+		 */
+		onProgress(event) {
+			let {currentTime, duration} = event,
+				{percentageTracked} = this.state;
+
+			/*
+			* Calculate the percentage played
+			*/
+			let percentagePlayed = Math.round(currentTime / duration * 100),
+				roundUpPercentage = Math.ceil(percentagePlayed / 25) * 25 || 25; // make sure that 0 is 25
+
+			/**
+			 * The following logic is applied:
+			 * Between 0% - 25% - Track Q1 mark
+			 * Between 25% - 50% - Track Q2 mark
+			 * Between 50% - 75% - Track Q3 mark
+			 * Between 75% - 100% - Track Q4 mark
+			 */
+			if (roundUpPercentage === 25 && !percentageTracked.Q1) {
+				this.trackQuarters(1, percentagePlayed);
+			} else if (roundUpPercentage === 50 && !percentageTracked.Q2) {
+				this.trackQuarters(2, percentagePlayed);
+			} else if (roundUpPercentage === 75 && !percentageTracked.Q3) {
+				this.trackQuarters(3, percentagePlayed);
+			} else if (roundUpPercentage === 100 && !percentageTracked.Q4) {
+				this.trackQuarters(4, percentagePlayed);
+			}
+
+			// Fire the call back of the onProgress
+			this.props.onProgress && this.props.onProgress(event);
+		}
+
+
+		/**
+		 * Method that tracks back to the "onEvent" call back to communicate a tracking of a quarter
+		 * @param {number} mark - The number of the quarter (1,2,3,4)
+		 * @param {number} percentagePlayed - The percentage played at the time of the tracking
+		 */
+		trackQuarters(mark, percentagePlayed) {
+			this.setState((prevState) => ({
+				percentageTracked: {
+					...prevState.percentageTracked,
+					[`Q${mark}`]: true
+				}
+			}));
+
+			this.onEvent({
+				type: PlayerEventTypes[`ON_PROGRESS_Q${mark}`],
+				percentagePlayed
+			});
+		}
+
+		/**
+		 * Event triggered when the fullscreen happens
+		 * @param {NativeEvent} event
+		 */
+		onEnterFullscreen(event) {
+			this.onEvent({'type': PlayerEventTypes.ON_ENTER_FULLSCREEN});
+			this.props.onEnterFullscreen && this.props.onEnterFullscreen(event);
+		}
+
+		/**
+		 * Event triggered when the user exists from the fullscreen
+		 * @param {NativeEvent} event
+		 */
+		onExitFullscreen(event) {
+			this.onEvent({'type': PlayerEventTypes.ON_EXIT_FULLSCREEN});
+			this.props.onExitFullscreen && this.props.onExitFullscreen(event);
+		}
+
+		/**
```

Comment:
I think the document is not correct for this @raf

Reply:
@pankaj-pp  fixed.